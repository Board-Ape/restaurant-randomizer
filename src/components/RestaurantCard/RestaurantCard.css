.card-container {
  width: 150px;
  display: flex;
  justify-content: space-around;
  display: inline-block;
  background-color: #f7f7f7;
  width: 400px;
  height: 300px;
  padding: 30px;
  margin: 10px;
  border-radius: 20px;
  border: 1px solid black;
  overflow: scroll;
  border: 5px #313131 solid; }

.card-container:hover {
  border: 5px #C10101 solid;
  cursor: pointer; }

.restaurant-name {
  font-family: 'Berkshire Swash', sans-serif;
  font-size: 40px;
  margin-top: 0;
  color: #C10101; }

.cuisine {
  font-weight: 600;
  font-size: 18px;
  color: #232323; }

.rating {
  font-size: 18px;
  color: #C10101; }

.address {
  font-weight: 500;
  font-size: 18px;
  color: #232323; }

.favorite {
  height: 10px;
  width: 10px;
  background-color: blue; }

.selected {
  background-color: black; }
