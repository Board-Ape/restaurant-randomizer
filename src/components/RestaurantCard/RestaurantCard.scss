.card-container {
  width: 150px;
  display: flex;
  justify-content: space-around;
  display: inline-block;
  background-color: white;
  width: 200px;
  height: 300px;
  padding: 30px;
  margin: 10px;
  border-radius: 20px;
  border: 1px solid black;
}

.cards {
  width: 200px;
  height: 300px;
  overflow:: scroll;
}

.address {
  font-size: 12px;
  color: white;
}
