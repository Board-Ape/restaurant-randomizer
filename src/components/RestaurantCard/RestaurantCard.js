import React, { Component } from 'react';
import './RestaurantCard.css';
import PropTypes from 'prop-types';
import { addFavorite, deleteFavorite } from '../../actions';
import { connect } from 'react-redux';

export class RestaurantCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false
    };
  }

  filterFavorites = (index) => {
    let results = this.props.favorites.filter(favorites => favorites.name === index);
    return results;
  }

  handleFavorites = (restaurant) => {
    this.setState({active: true});
    if (!this.filterFavorites(restaurant.restaurants.name).length) {
      return this.props.addFavorite(restaurant);
    } else {
      this.setState({active: false});
      return this.props.deleteFavorite(restaurant);
    }
  }

  render() {
    return (
      <div className='card-container'>
        <div className='cards'>
          <span
            className={this.state.active === false ? 'favorite' : 'favorite selected'}
            onClick={() => this.handleFavorites(this.props)}>
            Favorite
          </span>
          <h2 className='rating'>Rating: {this.props.restaurants.data.Rating}</h2>
          <h1 className='restaurant-name'>{this.props.restaurants.name}</h1>
          <h2 className='cuisine'>Cuisine: {this.props.restaurants.data.Cuisines}</h2>
          <h2 className='address'>Address:</h2>
          <h2 className='address'>{this.props.restaurants.data.Address}</h2>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (store) => {
  return {
    favorites: store.favorites
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    addFavorite: (favorites) => dispatch(addFavorite(favorites)),
    deleteFavorite: (favorites) => dispatch(deleteFavorite(favorites))
  };
};

RestaurantCard.propTypes = {
  restaurants: PropTypes.object,
  name: PropTypes.string,
  data: PropTypes.object,
  Rating: PropTypes.string,
  Cuisines: PropTypes.string,
  Address: PropTypes.string,
  favorites: PropTypes.array,
  addFavorite: PropTypes.func,
  deleteFavorite: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(RestaurantCard);
