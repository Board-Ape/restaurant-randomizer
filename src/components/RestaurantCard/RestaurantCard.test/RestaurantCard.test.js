import React from 'react';
import RestaurantCard from '../RestaurantCard';
import { shallow } from 'enzyme';

const mockData = {
  name: "Punch and Daisy",
  data: {
    Cuisines: "Cafe, Coffee and Tea, Sandwich",
    Address: "105 Stuart St, Mullumbimby",
    Rating: "3.8"
  }
};

describe('RestaurantCard', () => {
  let renderedRestaurantContainer;

  beforeEach(() => {
    renderedRestaurantContainer = shallow(
      <RestaurantCard
        restaurants={mockData}
      />
    );
  });

  it('should exist', () => {
    expect(renderedRestaurantContainer).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(renderedRestaurantContainer).toMatchSnapshot();
  });

  it('should render cards', () => {
    const expected = ['name', 'rating', 'cuisine', 'address'].length;

    expect(renderedRestaurantContainer.find('h2').length).toEqual(expected);
  });

});
