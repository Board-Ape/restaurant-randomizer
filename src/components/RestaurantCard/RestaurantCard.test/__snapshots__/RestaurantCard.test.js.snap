// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantCard
    data={
      Object {
        "sam": "Singer",
        "status": "Sickly",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cards"
      >
        <h3 />
      </div>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3 />,
        "className": "cards",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cards"
        >
          <h3 />
        </div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3 />,
          "className": "cards",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
